src/CFamily.UnitTests/Analysis/CLangAnalyzerTests.cs

```diff
@@ -309,16 +311,20 @@ private static Mock<IRequestFactoryAggregate> CreateRequestFactory(string filePa
             IAnalysisStatusNotifier statusNotifier = null,
             ICFamilyIssueConverterFactory issueConverterFactory = null,
             IRequestFactoryAggregate requestFactory = null,
-            ILogger logger = null)
+            ILogger logger = null,
+            IFileSystem fileSystem = null,
```

Comment:
Add both FileSystem and ThreadingHandling to construction as this gives more controll over whats happening in the tests.